backend/routers/slack.router.js

```diff
@@ -24,85 +20,53 @@ if (process.env.NODE_ENV !== "test") {
 }
 
 //Finds Id number of channel
-router.get("/findId", (req, res) => {
-  publishMessage();
-  findEvent();
-  // findProject();
-});
+router.get("/scheduleMessage", findCurrentEvents, scheduleMessage);
 
-//uses Id number to send message to said channel
-router.post("/postMeeting/:id", (req, res) => {
-  publishMessage1();
-});
+async function scheduleMessage(req, res, next) {
+  let channel;
+  let startTime;
+  let zoomChannel;
+  let events = res.locals.events;
+  events.forEach((cur) => {
+    let channelUrlArray = cur.project.slackUrl.split("/");
+    channel = channelUrlArray[channelUrlArray.length - 1];
+    startTime = cur.startTime;
+    zoomChannel = cur.project.videoConferenceLink;
+  });
+  startTime = startTime / 1000 - 3600000;
+  let currentTime = Math.round(new Date().getTime() / 1000);
 
-async function findConversation(name) {
   try {
-    const result = await app.client.conversations.list({
-      token: process.env.SLACK_BOT_TOKEN,
+    const result = await app.client.chat.scheduleMessage({
+      // The token you used to initalize the app
+      token: process.env.SLACK_OAUTH_TOKEN,
+      channel: channel,
+      text: "Slack Message Publish SCHEDULED",
+      // Time to post message, in Unix Epoch timestamp format
+      post_at: startTime,
     });
 
-    for (var channel of result.channels) {
-      if (channel.name === name) {
-        conversationId = channel.id;
-
-        console.log("Found conversation ID: " + conversationId);
-        break;
-      }
-    }
+    // Print result
   } catch (error) {
     console.error(error);
   }
 }
 
-async function publishMessage(id, text) {
-  try {
-    const result = await app.client.chat.postMessage({
-      token: process.env.SLACK_BOT_TOKEN,
-      channel: process.env.SLACK_CHANNEL_ID,
-      text: "Slack Message Publish",
-    });
-
-    console.log(result);
-  } catch (error) {
-    console.error(error);
-  }
-}
-
-async function publishMessage1(id, text) {
-  try {
-    const result = await app.client.chat.postMessage({
-      token: process.env.SLACK_BOT_TOKEN,
-      channel: id,
-      text: text,
-    });
-
-    console.log(result);
-  } catch (error) {
-    console.error(error);
-  }
-}
-
-async function findEvent(req, res) {
-  Event.find({})
+function findCurrentEvents(req, res, next) {
```

Comment:
I think that this relates to my statement above. I think that we are using a lot of space writing to the response object. We can likely use this logic inside the other call without.

Reply:
@nickbeaird Please take a look at my response to your previous comment.
@Jonathanko52 skip this comment.

Reply:
@alex-anakin, I must again respectfully disagree. 

This comment that I made relates to the above statement in regards to the POST middleware, but is slightly different. The method here is findCurrentEvents, and stands out to me as a GET endpoint. I would assume that we are asking to return all events that are current and that could be an updated GET. This method however is using Express middleware with the use of the next variable in the method (see link below). The docs for Express Middleware suggest using Middleware for when you need to chain methods together across API endpoints, such as setting headers across a wide array of endpoints, or adding in verification for routes. This allows you to isolate your verification of endpoints away from the actual logic. I would again state that this method and the logic in it are best suited to be pulled together in the endpoint, and not used as Express middleware across files. 

https://expressjs.com/en/guide/using-middleware.html